external/build-external.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Builds all external sources. This clears the data/ folder here
+ * and allows all scripts in build/ to write new files there.
+ *
+ * This is intended for use by Cloud Build, or by site devs doing local work.
+ */
+
+// nb. This is important; we load dotenv here and pass environment down.
+// It's safe to call this inside the build scripts too, but we run them with
+// cwd=data/, so it can't find the env.
+require('dotenv').config();
+
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const childProcess = require('child_process');
+const crypto = require('crypto');
+
+async function run() {
+  let errors = 0;
+
+  const scripts = glob.sync('build/*.js', {cwd: __dirname});
+  scripts.sort(); // run in alphabetical order
+
+  const dataTarget = path.join(__dirname, 'data');
+  fs.rmSync(dataTarget, {recursive: true});
+  fs.mkdirSync(dataTarget, {recursive: true});
+
+  /** @type {childProcess.CommonExecOptions} */
+  const options = {cwd: dataTarget, stdio: 'inherit'};
+
+  for (const script of scripts) {
+    const r = path.join(__dirname, script);
+    console.info('> Running', r);
+    try {
+      childProcess.execFileSync('node', [r], options);
+    } catch (e) {
+      // We don't log the error here, as we're already getting STDERR piped above.
+      console.warn(`! Failed to execute "${script}" (${e.status})`);
+      ++errors;
+    }
+  }
+
+  // Determine the hash for everything in data/.
+  const h = crypto.createHash('sha256');
```

Comment:
Would prefer a different name so I know what `h` is later in the code. Same for `r` above. I think the only one letter variables I like are for incrementors (`i`) or `e` for error. Otherwise a more descriptive variable name would be appreciated.